src/caret_analyze/infra/lttng/lttng.py

```diff
@@ -368,23 +467,77 @@ def _parse_lttng_data(
             begin = events[0][LttngEventFilter.TIMESTAMP]
             end = events[-1][LttngEventFilter.TIMESTAMP]
 
+            init_events = []
+            run_events = []
+
+            # 全てのトレースイベントを init_events, run_eventsに振り分ける
+            init_event_names = set(Lttng._prioritized_init_events)
             for event in events:
+                event_name = event[LttngEventFilter.NAME]
+                if event_name in init_event_names:
+                    init_events.append(event)
+                else:   # MYK test
+                    run_events.append(event)
+
+            # init_events をソートする
+            import functools
+            init_events.sort(key=functools.cmp_to_key(Lttng._compare_init_event))
+            # 初期化系トレースイベントを処理する
+            handler.create_init_handler_map()
+            for event in init_events:
+                event_name = event[LttngEventFilter.NAME]
+                if event_name in handler.handler_map:
+                    handler_ = handler.handler_map[event_name]
+                    handler_(event)
+
+            # ランタイム系トレースイベントを処理する
+            handler.create_runtime_handler_map()
+
+            for event in run_events:
                 if len(event_filters) > 0 and \
                         any(not f.accept(event, common) for f in event_filters):
                     continue
                 if store_events:
                     events.append(event)
                 filtered_event_count += 1
                 event_name = event[LttngEventFilter.NAME]
-                handler_ = handler.handler_map[event_name]
-                handler_(event)
+                if event_name in handler.handler_map:
+                    handler_ = handler.handler_map[event_name]
+                    handler_(event)
             data.finalize()
             if len(event_filters) > 0:
                 print('filtered to {} events.'.format(filtered_event_count))
 
         events_ = None if len(events) == 0 else events
         return data, events_, begin, end
 
+    @staticmethod
+    def _compare_init_event(
+        event1: dict,
+        event2: dict,
+    ) -> int:
+        """
+        Comparison function in sort function.
+
+        Returns
+        -------
+        str
+            1 if less, 0 if equal, and 1 if greater
+
+        """
+        if event2['_timestamp'] < event1['_timestamp']:
+            return 1
+        if event2['_timestamp'] > event1['_timestamp']:
+            return -1
+        # same timestamp
+        if Lttng._prioritized_init_events.index(event2['_name']) < \
+                Lttng._prioritized_init_events.index(event1['_name']):
+            return 1
+        if Lttng._prioritized_init_events.index(event2['_name']) > \
+                Lttng._prioritized_init_events.index(event1['_name']):
+            return -1
```

Comment:
- I suggest to use constants instend of plain strings.
- Condition of `return 1` and `return -1` is opposite.

```suggestion
        if event2[LttngEventFilter.TIMESTAMP] < event1[LttngEventFilter.TIMESTAMP]:
            return -1
        if event2[LttngEventFilter.TIMESTAMP] > event1[LttngEventFilter.TIMESTAMP]:
            return 1
        # same timestamp
        if Lttng._prioritized_init_events.index(event2[LttngEventFilter.NAME]) < \
                Lttng._prioritized_init_events.index(event1[LttngEventFilter.NAME]):
            return -1
        if Lttng._prioritized_init_events.index(event2[LttngEventFilter.NAME]) > \
                Lttng._prioritized_init_events.index(event1[LttngEventFilter.NAME]):
            return 1
```

Reply:
This should be correct, but the if parts are in the order of "event2" and "event1", so maybe I misunderstood? If Mikino should I reverse the order?

Reply:
It's correct, no corrections needed

Reply:
As you said, my following suggestion and wrong.
> Condition of return 1 and return -1 is opposite.

Still, my following suggestion is still valid. I will put another comment for this.
> I suggest to use constants instend of plain strings.